pkg/test/framework/components/echo/config.go

```diff
@@ -72,6 +72,16 @@ type Config struct {
 	// ReadinessTimeout specifies the timeout that we wait the application to
 	// become ready.
 	ReadinessTimeout time.Duration
+
+	// Workloads contains the list of workloads config belonging to this echo service instance.
+	Workloads []WorkloadConfig
+}
+
+// WorkloadConfig is the config for a group of workloads (e.g. Kubernetes deployment).
+type WorkloadConfig struct {
```

Comment:
Also, should this be called "Deployment" rather than workload? I suspect that each of these may provide the config for 1-n workloads

Reply:
oh you mean the number of the replicas can be N in a single deployment? sure. just maybe the deployment is very k8s specific.

Reply:
kept as Workload for above reason. feel free to re-address if you feel strong about this.

Reply:
Maybe `Subset` would be more appropriate?